scripts/lib/install

```diff
@@ -205,6 +243,63 @@ case "$POSTGRESQL_VERSION" in
         ;;
 esac
 
+if [ -z "$PUSH_NOTIFICATIONS" ] && [ -z "$NO_PUSH_NOTIFICATIONS" ]; then
+    # Unless specified, we default to --no-push-notifications
+    NO_PUSH_NOTIFICATIONS=1
+fi
+
+# We set these to Python-style True/False string values, for easy
+# insertion into the settings.py file.
+if [ -n "$PUSH_NOTIFICATIONS" ]; then
+    SERVICE_MOBILE_PUSH="True"
+
+    # --push-notifications also enables SUBMIT_USAGE_STATISTICS, unless the user
+    # explicitly opted out by passing --no-submit-usage-statistics.
+    if [ -n "$NO_SUBMIT_USAGE_STATISTICS" ]; then
+        SERVICE_SUBMIT_USAGE_STATISTICS="False"
+    else
+        SERVICE_SUBMIT_USAGE_STATISTICS="True"
+    fi
+else
+    SERVICE_MOBILE_PUSH="False"
+    # SUBMIT_USAGE_STATISTICS without PUSH_NOTIFICATIONS wouldn't really
+    # make much sense.
+    SERVICE_SUBMIT_USAGE_STATISTICS="False"
+fi
+
+# If user asked for push notifications, prompt for ToS acceptance.
+if [ -n "$PUSH_NOTIFICATIONS" ]; then
+    echo
+    echo "You chose to register your server for the Mobile Push Notifications Service."
+    echo "Doing so will share basic metadata with the service's maintainers, including:"
+    echo
+    echo "* The server's configured hostname: $EXTERNAL_HOST"
+    echo "* The server's configured contact email address: $ZULIP_ADMINISTRATOR"
+    echo "* Basic metadata about each organization hosted by the server; see:"
+    echo "    <https://zulip.readthedocs.io/en/latest/production/mobile-push-notifications.html#uploading-basic-metadata>"
+    echo
+    echo "Use of this service is governed by the Zulip Terms of Service:"
+    echo "    <https://zulip.com/policies/terms>"
+    echo
+    read -r -p "Do you want to agree to the Zulip Terms of Service and proceed? [Y/n] " tos_prompt
+    echo
+
+    # Normalize the user’s response to lowercase.
+    case "${tos_prompt,,}" in
+        "" | y | yes)
+            echo "Great! Push notifications will be enabled; continuing with installation..."
+            PUSH_NOTIFICATIONS_SERVICE_TOS_AGREED=1
+            SERVICE_MOBILE_PUSH=True
```

Comment:
Seems potentially confusing that we set this here, as well as above on line 254 -- but this one is redundant, I think, since it's already guaranteed to be set, and we exit in the other branch?  And even if the exit in the other branch got dropped, it would still stay set to True because of line 254?

Reply:
oh yeah, this is a leftover line i missed to clean up.